framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -369,29 +370,40 @@ static AnnotatedTypeMirror getTypeAtLocation(
             // getLocationTypeADT. For other kinds of types, no work is required for an empty
             // type path.
             return type;
-        } else if (type.getKind() == TypeKind.NULL) {
-            return getLocationTypeANT((AnnotatedNullType) type, location);
-        } else if (type.getKind() == TypeKind.DECLARED) {
-            return getLocationTypeADT((AnnotatedDeclaredType) type, location);
-        } else if (type.getKind() == TypeKind.WILDCARD) {
-            return getLocationTypeAWT((AnnotatedWildcardType) type, location);
-        } else if (type.getKind() == TypeKind.ARRAY) {
-            return getLocationTypeAAT((AnnotatedArrayType) type, location);
-        } else if (type.getKind() == TypeKind.UNION) {
-            return getLocationTypeAUT((AnnotatedUnionType) type, location);
-        } else if (type.getKind() == TypeKind.INTERSECTION) {
-            return getLocationTypeAIT((AnnotatedIntersectionType) type, location);
-        } else {
-            ErrorReporter.errorAbort(
-                    "ElementAnnotationUtil.getTypeAtLocation: only declared types, "
-                            + "arrays, and null types can have annotations with location; found type: "
-                            + type
-                            + " (kind: "
-                            + type.getKind()
-                            + ") location: "
-                            + location);
-            return null; // dead code
         }
+        switch (type.getKind()) {
+            case NULL:
+                return getLocationTypeANT((AnnotatedNullType) type, location);
+            case DECLARED:
+                return getLocationTypeADT((AnnotatedDeclaredType) type, location);
+            case WILDCARD:
+                return getLocationTypeAWT((AnnotatedWildcardType) type, location);
+            case TYPEVAR:
+                if (TypesUtils.isCaptured((TypeVariable) type.getUnderlyingType())) {
```

Comment:
This is the work-around for Issue #1696. The other changes in this PR remove single-use static imports which made the code harder to read.